projects/packages/forms/src/blocks/contact-form/editor.scss

```diff
@@ -958,9 +966,19 @@
 			&.is-selected, &.has-placeholder {
 
 				.animated-label__label {
-					font-size: 0.75em;
```

Comment:
This PR is working as described. 

There's a bit of a trade off here, as large font sizes make inputs in the animated style almost unusable, but that's the case in trunk too.


Here's what I'm seeing on the frontend:

https://github.com/user-attachments/assets/773cad99-fd97-4dcf-9d91-b1ba777904a7


The only think I noticed was that, in the editor, my large labels appear to animate "downwards" when focussed:

https://github.com/user-attachments/assets/62c898c5-abc0-47d1-8df0-6be274ef002a


Here's trunk for comparison:

https://github.com/user-attachments/assets/0cbf720e-b7e4-4854-b2a0-63aff586b5d5

Reply:
Well spotted.

It's unusual, both the label block and its parent have the `is-selected` class and that causes the animation to trigger. I'm not sure how they're both considered selected at the same time.

I pushed a css fix for it.

Reply:
Thanks for updating. 

Testing again I think it might be rather something to do with a change I made and the introduction of `--jetpack--contact-form--animated-top-offset` in `useVariationStyleProperties()`.

I didn't take into account the existing 1.4em padding. 

This seems to work better:

```js
'--jetpack--contact-form--animated-top-offset': hasTopBorder
					? 'calc(var(--jetpack--contact-form--border-top-size) + var(--jetpack--contact-form--animated-left-offset) + 1.4em)'
					: '50%',
```

Reply:
The `1.4em` corresponds to the value in editor.scss:

```scss
			.jetpack-field__input,
			.jetpack-field__textarea,
			.jetpack-field-dropdown__toggle
			{
				padding-top: 1.4em;
				padding-left: var(--jetpack--contact-form--animated-left-offset);
			}
```

Reply:
Actually, only for input fields 😮‍💨 

```js
		if ( formStyle === FORM_STYLE.ANIMATED ) {
			const hasTopBorder =
				!! borderWidths?.borderTopWidth && parseInt( borderWidths?.borderTopWidth ) > 0;
			const inputOffset = inputBlockName === 'jetpack/input' ? '1.4em' : '0';
			styleSpecificCssVars = {
				// For the animated labels.
				'--jetpack--contact-form--animated-left-offset': '16px', // Probably can be removed or we use `--jetpack--contact-form--input-padding`
				'--jetpack--contact-form--animated-top-offset': hasTopBorder
					? `calc(var(--jetpack--contact-form--border-top-size) + var(--jetpack--contact-form--animated-left-offset) + ${ inputOffset })`
					: '50%',
			};
		}
```

I'll get a new branch up so it's easier to test.

Reply:
> I'll get a new branch up so it's easier to test.


Actually, I don't think we're going to get this label to float in the vertical centre of the input field properly where the borders are not equal. It works on trunk because you can't set individual border widths

Reply:
Ok, I can reproduce it with some thicker borders. I'll see if I can figure it out.

Reply:
https://github.com/Automattic/jetpack/pull/43633

Reply:
> Actually, I don't think we're going to get this label to float in the vertical centre of the input field properly where the borders are not equal. It works on trunk because you can't set individual border widths

I think it is possible with something like this:
```
top: calc(50% + (var(--jetpack--contact-form--border-top-size, 0px) * 0.5) - (var(--jetpack--contact-form--border-bottom-size, 0px) * 0.5));
```

I can put a PR together.

Reply:
Nice!!

Feel free to commandeer or close https://github.com/Automattic/jetpack/pull/43633

Thanks a lot!

Reply:
Oh, sorry, didn't see you'd made a PR, I'd already started working on making one.

Here it is - #43635

Reply:
All good! I'll take a look at yours since I didn't bother with centering non equal top and bottom